src/TorchSharp/NN/Utils/PackedSequence.cs

```diff
@@ -0,0 +1,97 @@
+                {
+                    /// <summary>
+                    /// A packed batch of variable length sequences.
+                    /// </summary>
+                    public class PackedSequence : IDisposable
+                    {
+                        /// <summary>
+                        /// Class wrapping PyTorch's packedsequence object reference.
+                        /// </summary>
+                        internal sealed class HType : SafeHandle
+                        {
+                            public HType(IntPtr preexistingHandle, bool ownsHandle)
+                                : base(IntPtr.Zero, ownsHandle)
+                            {
+                                SetHandle(preexistingHandle);
+                            }
+
+                            public override bool IsInvalid => handle == IntPtr.Zero;
+
+                            // This is just for marshalling
+                            internal HType() : base(IntPtr.Zero, true)
+                            {
+                            }
+
+                            [DllImport("LibTorchSharp")]
+                            private static extern void THSNN_PackedSequence_dispose(HType handle);
+
+                            protected override bool ReleaseHandle()
+                            {
+                                THSNN_PackedSequence_dispose(this);
+                                return true;
+                            }
+
+                            protected override void Dispose(bool disposing)
+                            {
+                                if (disposing) {
+                                    ReleaseHandle();
+                                }
+                            }
+                        }
+
+                        private HType handle;
+
+                        internal PackedSequence(IntPtr handle)
+                        {
+                            if (handle != IntPtr.Zero) {
+                                this.handle = new HType(handle, true);
+                            }
+                        }
+
+                        ~PackedSequence()
+                        {
+                            Dispose(false);
+                        }
+
+                        internal HType Handle => handle;
+
+                        /// <summary>
+                        ///   Releases the storage.
+                        /// </summary>
+                        public void Dispose()
+                        {
+                            Dispose(true);
+                            GC.SuppressFinalize(this);
+                        }
+
+                        /// <summary>
+                        ///   Implements the .NET Dispose pattern.
+                        /// </summary>
+                        protected virtual void Dispose(bool disposing)
+                        {
+                            if (disposing && handle != null && !handle.IsInvalid) {
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0 seems not to check `disposing`. Let me verify this.

Reply:
@tarekgh Removed `Dispose(bool)`. https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0 says `handle != null` can happen.

Should this be
```
                    [DllImport("LibTorchSharp")]
                    private static extern IntPtr THSNN_pack_sequence(IntPtr[] sequences, int sequences_len, bool enforce_sorted);
```
like
```
                    [DllImport("LibTorchSharp")]
                    private static extern PackedSequence.HType THSNN_pack_sequence(IntPtr[] sequences, int sequences_len, bool enforce_sorted);
```
to avoid leaking?

Reply:
I asked the same question below before I saw your comment :-)

I am seeing the returned value is used in `new torch.Tensor(res);`. Tensor will call [THSTensor_dispose](https://github.com/dotnet/TorchSharp/blob/49910eceb6a6ed3b99103a5c1782f6b5b802021a/src/TorchSharp/Tensor/Tensor.cs#L87) to dispose the handle. Is this right for disposing the `PackedSequence`?

Reply:
`Tensor` and `PackedSequence` are working similarly but `Tensor` is not using `SafeHandle`. I assume that `PacketSequence` owns the handle and disposes it. `pack_sequence()` returns `PackedSequence`, `pad_sequence()` returns `Tensor`.